src/BinSkim.Rules/PERules/BA2004.EnableSecureSourceCodeHashing.cs

```diff
@@ -194,75 +200,153 @@ public void AnalyzeNativeBinaryAndPdb(BinaryAnalyzerContext context)
 
                     if (sf.HashType != HashType.SHA256)
                     {
+                        // Adding the `MD5` or `SHA1` etc, comment delimited by a backquote/backtick (`) to the library name
+                        string libraryNameWithHashComment = string.Format($"{omDetails.Library} `{sf.HashType}`");
+                        var objectModuleDetailsWithHashComment = new ObjectModuleDetails(omDetails.Name, libraryNameWithHashComment, omDetails.CompilerName, omDetails.CompilerFrontEndVersion, omDetails.CompilerBackEndVersion, omDetails.RawCommandLine, omDetails.Language, omDetails.HasSecurityChecks, omDetails.HasDebugInfo);
+
                         if (!string.IsNullOrEmpty(record.Library))
                         {
-                            compilandsLibraryWithOneOrMoreInsecureFileHashes.Add(omDetails);
+                            if (sf.HashType != HashType.None)
+                            {
+                                compilandsLibraryWithOneOrMoreInsecureFileHashes.Add(objectModuleDetailsWithHashComment);
+                            }
+                            else
+                            {
+                                compilandsLibraryWithOneOrMoreNoFileHashes.Add(objectModuleDetailsWithHashComment);
+                            }
                         }
                         else
                         {
-                            compilandsBinaryWithOneOrMoreInsecureFileHashes.Add(omDetails);
+                            if (sf.HashType != HashType.None)
```

Comment:
>HashType [](http://example.com/codeflow?start=35&length=8)

You did not follow my guidance to organize this information by a dictionary keyed by HashType.

Let's hold on this one. I will take a closer look tomorrow to see if I can better understand why you chose the direction you did. If possible, I'd prefer a simpler solution with less new complexity.

I note that you didn't add any new unit tests to the tool, despite adding a new rules extension method.